Find a functionally equivalent code:android.service.dreams.DreamService.dispatchGenericMotionEvent:COMMENT
Method Modifier: public      
Comment:/**
 * {@inheritDoc}
 */

@@ -1,8 +1,8 @@
 {
     if (!mInteractive) {
         if (mDebug)
-            Slog.v(TAG, "Finishing on genericMotionEvent");
-        safelyFinish();
+            Slog.v(TAG, "Waking up on genericMotionEvent");
+        wakeUp();
         return true;
     }
     return mWindow.superDispatchGenericMotionEvent(event);

